java-checks/src/main/java/org/sonar/java/checks/ThreadRunCheck.java

```diff
@@ -53,6 +56,17 @@
 
   @Override
   protected void onMethodInvocationFound(MethodInvocationTree mit) {
+    Tree parent = mit.parent();
+    while(parent != null && !parent.is(Tree.Kind.METHOD)) {
+      parent = parent.parent();
+    }
+    if(parent != null) {
+      MethodTree methodTree = (MethodTree) parent;
+      Symbol.MethodSymbol symbol = methodTree.symbol();
+      if("run".equals(symbol.name()) && symbol.parameterTypes().isEmpty() && symbol.owner().type().isSubtypeOf("java.lang.Runnable")) {
```

Comment:
Reuse the same methodMatcher used as entry point to check that the parent is a `run` method, it's way simpler.
```
if (parent != null && RUNNABLE_RUN_METHOD_MATCHER.matches((MethodTree) parent)) {
  return;
}
```